app/controllers/homepage_controller.rb

```diff
@@ -45,8 +45,38 @@ def publish
     redirect_to show_popular_links_path
   end
 
+  def destroy
+    if @latest_popular_links.can_delete?
+      @latest_popular_links.delete ? flash[:success] = "Popular links draft deleted.".html_safe : flash[:danger] = application_error_message
+    else
+      flash[:danger] = cannot_delete_published_error_message
+    end
+  rescue StandardError => e
+    Rails.logger.error "Error #{e.class} #{e.message}"
+    flash[:danger] = application_error_message
+  ensure
+    redirect_to show_popular_links_path
+  end
+
+  def confirm_destroy
+    if @latest_popular_links.can_delete?
+      render "homepage/popular_links/confirm_destroy"
+    else
+      flash[:danger] = cannot_delete_published_error_message
+      redirect_to show_popular_links_path
+    end
+  end
+
 private
 
+  def cannot_delete_published_error_message
+    "Can't delete an already published edition. Please create a new edition to make changes.".html_safe
```

Comment:
We would need to capture these messages in the [copy review doc](https://docs.google.com/document/d/1r_z2JLF-5RHpuHLainh0u4-pYHhaayrKvaC-yxNE_pQ/edit).